packages/blade/src/components/Indicator/Indicator.tsx

```diff
@@ -0,0 +1,102 @@
+import type { ReactElement } from 'react';
+import { useCallback } from 'react';
+import { useTheme } from '~components/BladeProvider';
+import Box from '~components/Box';
+import Svg from '~components/Icons/_Svg';
+import Circle from '~components/Icons/_Svg/Circle';
+import { Text } from '~components/Typography';
+
+import type { Feedback } from '~tokens/theme/theme';
+import { getPlatformType, makeAccessible } from '~utils';
+
+type IndicatorCommonProps = {
+  /**
+   * Sets the color tone
+   *
+   * @default neutral
+   */
+  intent?: Feedback;
+
+  /**
+   * Size of the indicator
+   *
+   * @default medium
+   */
+  size?: 'xsmall' | 'small' | 'medium';
+};
+
+type IndicatorWithoutA11yLabel = {
+  /**
+   * A text label to show alongside the indicator dot
+   */
+  children: string;
+
+  /**
+   * a11y label for screen readers
+   */
+  accessibilityLabel?: string;
+};
+
+type IndicatorWithA11yLabel = {
+  /**
+   * a11y label for screen readers
+   */
+  accessibilityLabel: string;
+
+  /**
+   * A text label to show alongside the indicator dot
+   */
+  children?: string;
+};
+
+type IndicatorProps = IndicatorCommonProps & (IndicatorWithA11yLabel | IndicatorWithoutA11yLabel);
+
+type Dimensions = {
+  svgSize: string;
+  textSize: 'small' | 'medium';
+};
+
+const Indicator = ({
+  accessibilityLabel,
+  children,
+  size = 'medium',
+  intent = 'neutral',
+}: IndicatorProps): ReactElement => {
+  const { theme } = useTheme();
+
+  const fillColor = theme.colors.feedback.background[intent].highContrast;
+  const strokeColor = theme.colors.brand.gray[200];
+  const getDimension = useCallback((): Dimensions => {
```

Comment:
Does this need to be a function? 

Can we use React.useMemo instead and inline the logic?

Reply:
Yes, both memo / callback works